spec/api/keys/keys_spec.rb

```diff
@@ -656,17 +563,262 @@
+
+    # Here we can re-use the our primary user/clients, because they are nor members of
+    # other-org
+    let (:other_org_user_name) { "#{$other_org_name}-user" }
+    let (:other_org_user) {requestor(other_org_user_name, keys[:other_org_user][:private]) }
+    let (:other_org_client_name) { "#{$other_org_name}-client" }
+    let (:other_org_client) {requestor(non_org_client_name, keys[:other_org_client][:private]) }
+
+    let (:base_user_payload) do
+      {
+        "first_name" => "Do",
+        "middle_name" => "Not",
+        "last_name" => "Care",
+        "display_name" => "Really Do Not",
+        "password" => "client-password",
+      }
+
+    end
+    let (:org_admin_payload) do
+    {
+      "public_key" => keys[:org_admin][:public],
+      "username" => org_admin_name,
+      "email" => "#{org_admin_name}@#{org_admin_name}.com"
+    }
+    end
+    let (:org_user_payload) do
+      {
+        "public_key" => keys[:org_user][:public],
+        "username" => org_user_name,
+        "email" => "#{org_user_name}@#{org_user_name}.com"
+      }
+    end
+    let (:other_org_user_payload) do
+      {
+        "public_key" => keys[:other_org_user][:public],
+        "username" => other_org_user_name,
+        "email" => "#{other_org_user_name}@#{other_org_user_name}.com"
+      }
+    end
+    let (:org_client_payload) do
+      {
+        "public_key" => keys[:org_client][:public],
+        "name" => org_client_name,
+        "admin" =>false
+      }
+    end
+
+    let (:other_org_client_payload) do
+      {
+        "public_key" => keys[:other_org_client][:public],
+        "name" => other_org_client_name,
+        "admin" =>false
+      }
+    end
+    let (:unassociated_requestor) do
+      requestor(user['name'], user['private_key'])
+    end
+
+    before :all do
+      # Create other org. This org has no members other than those created automatically
+
+
+      post("#{platform.server}/organizations", superuser, :payload => $other_org_payload).should look_like({:status => 201})
+    end
+    after :all do
+      delete("#{platform.server}/organizations/#{$other_org_name}/clients/#{$other_org_name}-validator", superuser).should look_like({:status => 200})
+      delete("#{platform.server}/organizations/#{$other_org_name}", superuser).should look_like({:status => 200})
+    end
+
+    before :each do
+      # status checks here to provide a level of protection against having to chase down unexpected errors
+      # because our startup state isn't what we think it is.
+      post("#{platform.server}/users", superuser, :payload => base_user_payload.dup.merge(org_admin_payload)).should look_like({:status => 201} )
+      post("#{platform.server}/users", superuser, :payload => base_user_payload.dup.merge(org_user_payload)).should look_like({:status => 201} )
+      post("#{platform.server}/users", superuser, :payload => base_user_payload.dup.merge(other_org_user_payload)).should look_like({:status => 201} )
```

Comment:
Too many cases of of 'why are my tests failing' ? Oh, this precondition never completed succesfully, but I didn't know it.  

I agree in theory - your tests should be the only thing testing anything.  Practically though, I've been bitten too many times by silently failed pre- and post-conditions (including here, before I added the checks).  

So while I'm not checking extensively  - response bodies, etc - I do tend to check bare  minimum necessary to be able to proceed with running the actual test, when the effects of the change are required to make the test successful & meaningful. 


And what's actually nice is that pedant (now? not sure if it always did) points out when it's a before/after failure, so you don't waste time looking at the individual tests when it happens.